oauth2/handler.go

```diff
@@ -1322,12 +1327,19 @@ func (h *Handler) writeAuthorizeError(w http.ResponseWriter, r *http.Request, ar
 
 // updateSessionWithRequest takes a session and a fosite.request as input and returns a new session.
 // If any errors occur, they are logged.
-func (h *Handler) updateSessionWithRequest(ctx context.Context, session *flow.AcceptOAuth2ConsentRequest, flow *flow.Flow, r *http.Request, request fosite.Requester) (*Session, error) {
-	for _, scope := range session.GrantedScope {
+func (h *Handler) updateSessionWithRequest(
+	ctx context.Context,
+	consent *flow.AcceptOAuth2ConsentRequest,
+	flow *flow.Flow,
+	r *http.Request,
```

Comment:
**Non-blocking**

Just a random question. I just realized the argument `r` was only used for logging. Is it possible we could pass it inside the `ctx`?

Reply:
The context [doc](https://pkg.go.dev/context) says:

> Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions. 

In general, context key-value storage is untyped (it is `any -> any`), which means that you need to check and cast the variable every time you wish to extract it. This is not a great use-case for using the context.